syntax = "proto3";

package gr;

message Data {
  string s = 1;
  bytes b = 2;
  string json = 3;
}

message ServiceCallParam {
  string name = 1;
  Data data = 2;
}

message ServiceCall {
  string namespace = 1;
  string service = 2;
  string version = 3;
  string method = 4;
  repeated ServiceCallParam parameters = 5;
}

service ServiceInvoker {
  /**
   * Invoke a single method.
   */
  rpc InvokeMethod(ServiceCall) returns (Data) {}
}
